FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Tasks/TaskExecutor.cs

```diff
@@ -52,10 +52,10 @@ public sealed class TaskExecutor : ITaskExecutor
         // Add cancelling
         public async Task<TaskResult> ExecuteAsync(
             TaskContext taskContext,
-            IProgress<TaskContext> progress,
+            JobProgressUpdater progressUpdater,
             CancellationToken cancellationToken = default)
         {
-            int processedCount = 0;
+            int pageCount = 0;
```

Comment:
Does this `pageCount` refer to the total response pages/times from FHIR Server?

Reply:
Yes, here we want to report progress on a frequency proportional to FHIR API request counts because the resource count returned is not stable in each request